src/promise.js

```diff
@@ -88,6 +88,7 @@ function Promise(executor) {
         // by enclosed function which is added later
         // by build process (tools/build.js!buildBrowser)
         this.currentHadronTest = _global.currentHadronTest;
+        this.creationStack     = new Error().stack;
```

Comment:
updated to user the internal function that captures stack - it does so only when longStackTrace is enabled.